### PR TITLE
add support for firewalld

### PR DESCRIPTION
**Pull request by Sispheor**
Add a support for firewalld.

The variable `type` leave an opened window for futur support like iptables.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -24,3 +24,12 @@ prometheus_node_exporter_enabled_collectors:
...
   'log.level': 'info'
+
+# if the node is behind a firewall
+# active: set to True to open the firewall on the "listen_address"
+# type: only "firewalld" supported so far
+# listen_address: must be the same as the "web.listen-address". By default 9100
+prometheus_node_exporter_firewall:
+  active: False
+  type: firewalld
+  listen_address: 9100
```
Comment (UnderGreen) [inline defaults/main.yml]: `listen_port` because it is a port, not an address

**Comment by UnderGreen** [comment]
Also, I didn't understand from PR what the problem you try to resolve.

**Comment by Sispheor** [comment]
Fixed.

And the problem I want to resolve is the fact that the export listen on a TCP port. And in some case, the node use a firewall software like iptables or firewalld.
In the case, the user just need to precise it and a rule will be added automatically.

**Comment by UnderGreen** [comment]
I understood our approach, but think you try to solve the problem from the wrong side. My approach here - firewall rules should be set by firewall ansible role, not here.

**Comment by Sispheor** [comment]
Actually, there is no fixed in stone practice. I've deployed a lot of community based Ansible roles and in most of case they are written to be used in standalone. This mean that they handle all parameters on a system that are needed to make the service working. Like adding a SElinux or firewall rules. Because when we talk about a "role", we expect that this one is actually available when the script has been applied.
That's why have included the feature.

It's your role, so up to you to tell me which "best practice" you want to follow.

**Pull request closed without merging**